internal/fortran/fortran.go

```diff
@@ -0,0 +1,422 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package fortran provides utilities to help interface to fortran routines.
+//
+// The utility functions provided in fortran are not intended to be used generally
+// and provide very limited friendliness. They are intended to be used in development
+// of Go ports of FORTRAN routines.
+package fortran
+
+import (
+	"github.com/gonum/blas"
+	"github.com/gonum/blas/blas64"
+)
+
+// TransCharacter returns the FORTRAN character for the given transpose operation.
+func TransCharacter(trans blas.Transpose) byte {
+	switch trans {
+	case blas.NoTrans:
+		return 'N'
+	case blas.Trans:
+		return 'T'
+	case blas.ConjTrans:
+		return 'C'
+	default:
+		panic("fortran: bad BLAS trans")
+	}
+}
+
+// UploCharacter returns the FORTRAN character for the given uplo state.
+func UploCharacter(uplo blas.Uplo) byte {
+	switch uplo {
+	case blas.Upper:
+		return 'U'
+	case blas.Lower:
+		return 'L'
+	case blas.All:
+		return 'A'
+	default:
+		panic("fortran: bad BLAS uplo")
+	}
+}
+
+// DiagCharacter returns the FORTRAN character for the given diagonal state.
+func DiagCharacter(diag blas.Diag) byte {
+	switch diag {
+	case blas.Unit:
+		return 'U'
+	case blas.NonUnit:
+		return 'N'
+	default:
+		panic("fortran: bad BLAS diag")
+	}
+}
+
+// SideCharacter returns the FORTRAN character for the given side state.
+func SideCharacter(side blas.Side) byte {
+	switch side {
+	case blas.Left:
+		return 'L'
+	case blas.Right:
+		return 'R'
+	default:
+		panic("fortran: bad BLAS side")
+	}
+}
+
+// General is a column major general matrix.
+type General blas64.General
+
+// FromRowMajorGeneral returns a column major general matrix
+// with the same dimensions and data elements as the row major a.
+func FromRowMajorGeneral(a blas64.General) General {
```

Comment:
I had some ideas about changing the API that will have this as a natural consequence. I'll push them in a second.